### PR TITLE
add cluster persister for ipam resource refactoring

### PR DESCRIPTION
**Pull request by xh3b4sd**
Same as https://github.com/giantswarm/aws-operator/pull/1755 but for persisting and not checking.

### CODE REVIEW

File: service/controller/clusterapi/v29/network/cluster_persister.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+type ClusterPersisterConfig struct {
+	CMAClient clientset.Interface
+	Logger    micrologger.Logger
+}
+
+type ClusterPersister struct {
```
Comment (MarcelMue) [inline service/controller/clusterapi/v29/network/cluster_persister.go]: WDYT about naming this `ClusterStatusPersister` - it's still a weird naming but it makes it a bit clearer for me where and what gets persisted.

Comment (xh3b4sd) [reply service/controller/clusterapi/v29/network/cluster_persister.go]: I don't like it but when this is important to you we can go with it.

Comment (MarcelMue) [reply service/controller/clusterapi/v29/network/cluster_persister.go]: For me it gives very little information about what it does when reading code - maybe I need to get into the codebase more first but when reading it there were multiple questions in my head (What is cluster exactly, where does it persist and so on).

Comment (xh3b4sd) [reply service/controller/clusterapi/v29/network/cluster_persister.go]: We talked. We keep the names short. This here is WIP and when we put everything together in the `ipam` resource we take care of decent godoc for all the different implementations.

**Pull request merged**